src/webgl/p5.RendererGL.js

```diff
@@ -1724,54 +1726,41 @@ p5.RendererGL = class RendererGL extends Renderer {
   }
 
   /*
-   * selects which fill shader should be used based on renderer state,
-   * for use with begin/endShape and immediate vertex mode.
+   * This method will handle both image shaders and
+   * fill shaders, returning the appropriate shader
+   * depending on the current context (image or shape).
    */
-  _getImmediateFillShader() {
-    const fill = this.userFillShader;
-    if (this._useNormalMaterial) {
-      if (!fill || !fill.isNormalShader()) {
-        return this._getNormalShader();
+  _getFillShader() {
+    // If a filter like BLUR is applied, disable imageShader
+    if (this._drawingFilter) {
+      return this._getLightShader(); // Ignore user-defined imageShader
+    }
+    // If drawing an image, use the image shader or light shader
+    else if (this._drawingImage) {
+      if(this.userImageShader){
```

Comment:
like, instead of adding an extra condition can we just do something like

```js
if(this.userImageShader && !this._drawingFIlter){
// your code
} else{
return this._getLightShader();
}
```